analysis/wwz/make_datacards.py

```diff
@@ -252,27 +252,28 @@ def handle_per_year_systs_for_fr(in_dict,year_name,do_jec27):
 #   - Replace the value with SMALL
 #   - And add |value| to the stat error to be more conservative
 #   - Shift the up/down variations to be centered around SMALL (does not touch stat uncertainty on up/down)
-def handle_negatives(in_dict):
+def handle_negatives(in_dict,zero_low_mc):
     out_dict = copy.deepcopy(in_dict)
     for cat in in_dict:
         for proc in in_dict[cat]["nominal"]:
             val = in_dict[cat]["nominal"][proc][0]
             var = in_dict[cat]["nominal"][proc][1]
             if val <= 0:
-                #!!!OLD METHOD FOR HANDLING LOW YIELD BINS!!!
-                #print(f"WARNING: Process \"{proc}\" in cat \"{cat}\" is negative ({val}), replacing with {SMALL} and shifting up/down systematic variations accordingly.")
-                #out_dict[cat]["nominal"][proc][0] = SMALL
-                #out_dict[cat]["nominal"][proc][1] = (abs(val) + np.sqrt(var))**2
-                #for syst in out_dict[cat]:
-                #    if syst == "nominal": continue # Already handled this one
-                #    syst_var_orig = out_dict[cat][syst][proc][0] # Dont bother messsing with mc stat error on the syst variation
-                #out_dict[cat][syst][proc][0] = (syst_var_orig - val) + SMALL # Center around SMALL
-                print(f"WARNING: Process \"{proc}\" in cat \"{cat}\" is negative ({val}), replacing with {SMALL} and setting variations to 1/1.")
-                out_dict[cat]["nominal"][proc][0] = SMALL
-                out_dict[cat]["nominal"][proc][1] = 0
-                for syst in out_dict[cat]:
-                    if syst == "nominal": continue # Already handled this one
-                    out_dict[cat][syst][proc][0] = SMALL
+                if not zero_low_mc:
+                    print(f"WARNING: Process \"{proc}\" in cat \"{cat}\" is negative ({val}), replacing with {SMALL} and shifting up/down systematic variations accordingly.")
+                    out_dict[cat]["nominal"][proc][0] = SMALL
+                    out_dict[cat]["nominal"][proc][1] = (abs(val) + np.sqrt(var))**2
+                    for syst in out_dict[cat]:
+                        if syst == "nominal": continue # Already handled this one
+                        syst_var_orig = out_dict[cat][syst][proc][0] # Dont bother messsing with mc stat error on the syst variation
+                        out_dict[cat][syst][proc][0] = (syst_var_orig - val) + SMALL # Center around SMALL
+                elif zero_low_mc:
```

Comment:
I think this might be more straightforward as an if and and else. (The elif implies, at least to me, that there are more than 2 options here).

I would also swap the order so that it goes "if zero_low_mc ... else". That way the zero low mc part is more explicit.